Wise/src/main/java/co/cask/cdap/apps/wise/BounceCountsMapReduce.java

```diff
@@ -166,11 +166,11 @@ protected void cleanup(Context context) throws IOException, InterruptedException
    * Partition {@link LogInfo} using the IP address only so that a reducer gets all URI visited
    * by a particular IP. This is needed to detect which page a user last visited.
    */
-  public static class NaturalKeyPartitioner extends Partitioner<LogInfo, Text> {
+  public static class NaturalKeyPartitioner extends Partitioner<LogInfo, IntWritable> {
     @Override
-    public int getPartition(LogInfo logInfo, Text value, int numPartitions) {
+    public int getPartition(LogInfo logInfo, IntWritable value, int numPartitions) {
       int hash = logInfo.getIp().hashCode();
-      int partition = hash % numPartitions;
```

Comment:
Because it uses [String.hashcode()](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/String.java#String.hashCode%28%29) which can overflow.